lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -5296,16 +5296,35 @@ FIRCircuitParser::parseModuleBody(DeferredModuleToParse &deferredModule) {
 
   // Convert print-encoded verifications after parsing.
 
-  // It is dangerous to modify IR in the walk, so accumulate printFOp to
-  // buffer.
-  SmallVector<PrintFOp> buffer;
-  deferredModule.moduleOp.walk(
-      [&buffer](PrintFOp printFOp) { buffer.push_back(printFOp); });
-
-  for (auto printFOp : buffer) {
-    auto result = circt::firrtl::foldWhenEncodedVerifOp(printFOp);
-    if (failed(result))
-      return result;
+  {
+    // Gather PrintFOps and then process to avoid walking while mutating.
+    SmallVector<PrintFOp> buffer;
+    deferredModule.moduleOp.walk(
```

Comment:
This walk seems unnecessary.  Should we be collecting printfs during parse in something in the statement parser?  I realize this is a pre-existing condition.

Reply:
Agreed.  Inclined to leave as-is for now with plan to phase all of this out soon (and so would rather not thread that through just to remove it).